### PR TITLE
OPENEUROPA-3278: Expose Europa Search as block.

### PR DESCRIPTION
**Pull request by sergepavle**
### Description

[Insert description here]

### Change log

- Added:
- Changed:
- Deprecated:
- Removed:
- Fixed:
- Security:

### Commands

```sh
[Insert commands here]

```

### CODE REVIEW

File: modules/oe_webtools_europa_search/tests/src/Kernel/EuropaSearchBlockTest.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      'label_display' => '0',
+    ];
+    foreach (['fr', 'en'] as $langcode) {
+      \Drupal::configFactory()->getEditable('system.site')->set('default_langcode', $langcode)->save();
+      \Drupal::service('kernel')->rebuildContainer();
+      $plugin = $this->container->get('plugin.manager.block')
+        ->createInstance('europa_search', $config);
+      $render = $plugin->build();
+      // Make sure the block has the required loaders.
+      $this->assertEquals(['oe_webtools/drupal.webtools-smartloader'], $render['content']['#attached']['library']);
```
Comment (brummbar) [inline modules/oe_webtools_europa_search/tests/src/Kernel/EuropaSearchBlockTest.php:56]: Let's also check that the cache context is there.

Comment (sergepavle) [reply modules/oe_webtools_europa_search/tests/src/Kernel/EuropaSearchBlockTest.php:57]: Done

File: modules/oe_webtools_europa_search/tests/src/Kernel/EuropaSearchBlockTest.php [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  protected function setUp() {
+    parent::setUp();
+    $this->installEntitySchema('user');
+    $this->installSchema('system', 'sequences');
+    $this->installConfig(['system']);
+    ConfigurableLanguage::createFromLangcode('fr')->save();
+  }
+
+  /**
+   * Test europa search block rendering.
```
[the remark this answers is not in the document]
Comment (sergepavle) [reply modules/oe_webtools_europa_search/tests/src/Kernel/EuropaSearchBlockTest.php]: fixed

File: modules/oe_webtools_europa_search/tests/src/Kernel/EuropaSearchBlockTest.php [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+   */
+  public function testSocialShareBlockRendering(): void {
```
[the remark this answers is not in the document]
Comment (sergepavle) [reply modules/oe_webtools_europa_search/tests/src/Kernel/EuropaSearchBlockTest.php]: fixed

File: modules/oe_webtools_europa_search/src/Plugin/Block/EuropaSearchBlock.php [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+  /**
+   * The language manager.
+   *
+   * @var \Drupal\Core\Language\LanguageManagerInterface
+   */
+  protected $languageManager;
+
+  /**
+   * Creates a LocalActionsBlock instance.
```
[the remark this answers is not in the document]
Comment (sergepavle) [reply modules/oe_webtools_europa_search/src/Plugin/Block/EuropaSearchBlock.php]: fixed

File: modules/oe_webtools_europa_search/src/Plugin/Block/EuropaSearchBlock.php [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+use Drupal\Core\Language\LanguageManagerInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides an europa search webtools widget.
+ *
+ * @Block(
+ *   id = "europa_search",
+ *   admin_label = @Translation("Europa Search"),
```
[the remark this answers is not in the document]
Comment (sergepavle) [reply modules/oe_webtools_europa_search/src/Plugin/Block/EuropaSearchBlock.php]: fixed

File: modules/oe_webtools_europa_search/src/Plugin/Block/EuropaSearchBlock.php [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+use Drupal\Core\Language\LanguageInterface;
+use Drupal\Core\Language\LanguageManagerInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides an europa search webtools widget.
+ *
+ * @Block(
+ *   id = "europa_search",
```
[the remark this answers is not in the document]
Comment (sergepavle) [reply modules/oe_webtools_europa_search/src/Plugin/Block/EuropaSearchBlock.php]: done

File: modules/oe_webtools_europa_search/src/Plugin/Block/EuropaSearchBlock.php [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+use Drupal\Component\Serialization\Json;
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Cache\CacheableMetadata;
+use Drupal\Core\Language\LanguageInterface;
+use Drupal\Core\Language\LanguageManagerInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides an europa search webtools widget.
```
[the remark this answers is not in the document]
Comment (sergepavle) [reply modules/oe_webtools_europa_search/src/Plugin/Block/EuropaSearchBlock.php]: done

File: modules/oe_webtools_europa_search/oe_webtools_europa_search.info.yml [hunk 8]
```diff
@@ -0,0 +1,7 @@
+name: OpenEuropa Webtools Europa Search
+description: Provides a block that places the europa search script.
```
[the remark this answers is not in the document]
Comment (sergepavle) [reply modules/oe_webtools_europa_search/oe_webtools_europa_search.info.yml]: done

**Pull request merged**